components/pkg-export-kubernetes/src/main.rs

```diff
@@ -0,0 +1,178 @@
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#[macro_use]
+extern crate clap;
+extern crate env_logger;
+extern crate habitat_core as hcore;
+extern crate habitat_common as common;
+extern crate handlebars;
+#[macro_use]
+extern crate serde_json;
+#[macro_use]
+extern crate log;
+
+use clap::App;
+use handlebars::Handlebars;
+use std::result;
+use std::str::FromStr;
+use std::io::prelude::*;
+use std::io;
+use std::fs::File;
+
+use hcore::PROGRAM_NAME;
+use hcore::env as henv;
+use hcore::package::PackageIdent;
+use common::ui::{Coloring, UI, NOCOLORING_ENVVAR, NONINTERACTIVE_ENVVAR};
+
+// Synced with the version of the Habitat operator.
+pub const VERSION: &'static str = "0.1.0";
+
+// Kubernetes manifest template
+const MANIFESTFILE: &'static str = include_str!("../defaults/KubernetesManifest.hbs");
+
+fn main() {
+    env_logger::init().unwrap();
+    let mut ui = ui();
+    if let Err(e) = start(&mut ui) {
+        let _ = ui.fatal(e);
+        std::process::exit(1)
+    }
+}
+
+fn ui() -> UI {
+    let isatty = if henv::var(NONINTERACTIVE_ENVVAR)
+        .map(|val| val == "true")
+        .unwrap_or(false)
+    {
+        Some(false)
+    } else {
+        None
+    };
+    let coloring = if henv::var(NOCOLORING_ENVVAR)
+        .map(|val| val == "true")
+        .unwrap_or(false)
+    {
+        Coloring::Never
+    } else {
+        Coloring::Auto
+    };
+    UI::default_with(coloring, isatty)
+}
+
+fn start(_ui: &mut UI) -> result::Result<(), String> {
+    let m = cli().get_matches();
+    debug!("clap cli args: {:?}", m);
+    let count = m.value_of("COUNT").unwrap_or("1");
+    let topology = m.value_of("TOPOLOGY").unwrap_or("standalone");
+    let (group, add_group) = match m.value_of("GROUP") {
+        Some(g) => (g, true),
+        None => ("default", false),
+    };
+    let (config_secret_name, add_config_secret_name) =
+        match m.value_of("CONFIG_SECRET_NAME") {
+            Some(s) => (s, true),
+            None => ("user-toml-secret", false),
+        };
+    let (ring_secret_name, add_ring_secret_name) =
+        match m.value_of("RING_SECRET_NAME") {
+            Some(s) => (s, true),
+            None => ("SECRET_NAME", false),
+        };
+    // clap_app!() ensures that we do have the mandatory args so unwrap() is fine here
+    let pkg_ident_str = m.value_of("PKG_IDENT").unwrap();
+    let pkg_ident = match PackageIdent::from_str(pkg_ident_str) {
+        Ok(pi) => pi,
+        Err(e) => return Err(format!("{}", e)),
+    };
+    let image = m.value_of("IMAGE").unwrap_or(pkg_ident_str);
+
+    let json = json!({
+        "metadata_name": pkg_ident.name,
+        "image": image,
+        "count": count,
+        "service_topology": topology,
+        "add_service_group": add_group,
+        "service_group": group,
+        "add_config_secret_name": add_config_secret_name,
```

Comment:
Can the template operate on whether `service_group` is a `Some` or `None`, instead of having to pass a boolean in as well?

Reply:
didn't realize you can pass an option type to json. I'll try.. would be much better indeed.

Reply:
it works \o/